### ISSUE TITLE
Model and parameter related

### ISSUE THREAD

**Issue by mgao97** [SongtaoLiu0823/LAGNN#10]
Hi~

Thank you for your great work! I have two questions about it. 

1)The first one is about the structure of the VAE model. Why the encoder and decoder are both based on the MLP layers?

2)The second one is about the parameters. For example, could you please explain the meanings of the hidden size, latent size, and tem within the lagcn.py according to the repo? Why are they set as current values? How to set them properly for different datasets?

Thank you and look forward to your reply.

**Comment by SongtaoLiu0823** [comment]
Hi,

1) We follow the traditional setting of VAE, so we use MLP layers.
2) The hidden size is the number of parameters of MLP layers. The latent size is the number of latent variables. We use [cvae_tune.py](https://github.com/SongtaoLiu0823/LAGNN/blob/main/Citation/cvae_tune.py) to get the values.

**Comment by mgao97** [comment]
Thank you for your information.

I have another question about the way to pre-training the cvae model. 
I tried to retrain the cvae model for Cora dataset according to the cvae_generate_citation.py, but I found that it does not work and I could not obtain a better pre-trained model. Is there anything I should be aware of or I have missed? 

Your reply helps me a lot!

**Comment by SongtaoLiu0823** [comment]
Hi,

We use the cvae_tune.py to get the hyper parameters. We provide cvae_generate_citation.py to show how to use our codes.

**Comment by mgao97** [comment]
Hi,

Thank you so much. So if I want to pretrain the cvae model for another dataset which are not included in this repo. 

So should I first use the cvae_tune.py to get the hyperparameters for the cvae model? Then use another script like "cvae_generated_datasetname.py" to obtain the pre-trained cvae model. I hope I am getting this right :)

Thank you and look forward to your reply.

**Comment by SongtaoLiu0823** [comment]
Yes, you are correct.

**Comment by mgao97** [comment]
I see, thank you for your information and patience.

**Comment by mgao97** [comment]
Hi,

One more question about the results for the Cora dataset from the cvae_tune.py, where the values of "print(np.mean(all_maxVal1Acc), np.mean(all_maxVal1Acc_Val2Acc))" are 0.8166800000000001 and 0.7996400000000001, respectively. All hyper parameters are set by following the appendix of your paper. 

Are these values the same as that of yours? I think that these values are not ideal, and they are lower than the performance on the test set.

Any suggestions for better results?

**Comment by SongtaoLiu0823** [comment]
We use np.mean(all_maxVal1Acc_Val2Acc)) to select the hyper parameters. And we try many hyper parameters, and get the best model based on np.mean(all_maxVal1Acc_Val2Acc)). Then use it for downstream tasks.

**Comment by mgao97** [comment]
> We use np.mean(all_maxVal1Acc_Val2Acc)) to select the hyper parameters. And we try many hyper parameters, and get the best model based on np.mean(all_maxVal1Acc_Val2Acc)). Then use it for downstream tasks.

Got it! Thank you for your message.

Could you provide the best value of _np.mean(all_maxVal1Acc_Val2Acc))_ or its approximation? I would like to use it as a guide for the best model. Thank you so much!

**Comment by SongtaoLiu0823** [comment]
I will try my best to find my log file.

**Comment by mgao97** [comment]
Thanks a lot!

**Comment by SongtaoLiu0823** [comment]
For Cora, it's 0.82312, For Citeseer, it's 0.72888, For Pubmed, it's 0.80416.

**Comment by mgao97** [comment]
Thanks for your information~
